src/verilog/verilog_typecheck_expr.cpp

```diff
@@ -2288,16 +2290,16 @@ exprt verilog_typecheck_exprt::convert_trinary_expr(ternary_exprt expr)
     if(op1 >= width + offset)
     {
       auto padding_width = op1 - (width + offset) + 1;
-      auto padding =
-        from_integer(0, unsignedbv_typet{padding_width.to_ulong()});
-      auto new_type =
-        unsignedbv_typet{(get_width(op0.type()) + padding_width).to_ulong()};
+      auto padding = from_integer(
+        0, unsignedbv_typet{numeric_cast_v<unsigned>(padding_width)});
+      auto new_type = unsignedbv_typet{
+        numeric_cast_v<unsigned>(get_width(op0.type()) + padding_width)};
       expr.op0() = concatenation_exprt(padding, expr.op0(), new_type);
     }
 
     // Part-select expressions are unsigned, even if the
     // entire expression is selected!
-    auto expr_type = unsignedbv_typet((op1 - op2).to_ulong() + 1);
+    auto expr_type = unsignedbv_typet{numeric_cast_v<unsigned>(op1 - op2 + 1)};
```

Comment:
Here and above: is it really ok to use the platform-dependent `unsigned` type?

Reply:
Changed to `std::size_t` -- however, I think I really want to change the constructor of the bitvector types to `mp_integer`. Client-code should not have to deal with this.

Reply:
Yes, indeed.